File: adventofcode.com/2024/3/1.rb
result = 0

File.open(File.dirname(__FILE__) + "/input.txt", "r") do |f|
  f.each_line do |line|
    line.scan(/mul\(\d+,\d+\)/).each do |operation|
      values = operation[4..-2].split(',')
      result += Integer(values[0]) * Integer(values[1])
    end
  end
end

puts(result)

File: adventofcode.com/2024/03/1.py
import os
import re

result = 0

with open(os.path.dirname(__file__) + '/input.txt', mode='r') as f:
    for raw_line in f.readlines():
        for operation in re.findall(r'mul\(\d+,\d+\)', raw_line):
            values = operation[4:-1].split(',')
            result += int(values[0]) * int(values[1])

print(result)  # 170778545

